### ISSUE TITLE
Conflicting symbol for `config const` prevents compile-time configuration

### ISSUE THREAD

**Issue by BryantLam** [chapel-lang/chapel#11782]
Related to #11643.

This code works:
```chapel
config const CV = 1;
config param PV = 2;
writeln((CV, PV)); // (1, 2)
```
if I compile with `chpl --set CV=11 --set PV=12`, I get the expected output `(11, 12)`.

Now, with the code from #11643,
```chapel
module M {
  config const CV = 1;
  config param PV = 2;
}
config const CV = 3;
config param PV = 4;
writeln((M.CV, M.PV, CV, PV)); // (1, 2, 3, 4)
```

I have three items:

#### Issue: Cannot disambiguate CV at compile time
Similar to issue #11643 on PV, I cannot disambiguate CV at compile-time to set a runtime constant value.
- `chpl --set CV=11` yields compilation error `error: Ambiguous config param or type name (ConstValue)`
- `chpl --set Module.CV=11` yields compilation error `error: Trying to set unrecognized config param 'Module.ConstValue' via -s flag`

#### Issue: Are the error messages accurate?
From the previous, the error messages say "config param" when this variable is a "config const".

#### Question: Why is a `const` modifiable at compile time?
Why is this `const` variable able to be set at both compile time and runtime? It kind of makes sense to me, but it's also kind of not what I expected. My guess would be that it doesn't make much difference to set a value at compile-time vs. runtime other than the runtime value being more flexible (with less potential for optimization) and cannot be used in param-only cases.

`chpl version 1.19.0 pre-release (0001e1ca6a)`

**Comment by bradcray** [comment]
> Issue: Cannot disambiguate CV at compile time

As you point out, I think this is another form of issue #11643.

> Issue: Are the error messages accurate?

Good point.  I think you're correct that they're obviously not.  I suspect that only config params and types were supported originally and that they were later extended to support config consts and vars without updating the error message (?).

> Question: Why is a const modifiable at compile time?

Your intuition is correct that this is a bit surprising and (arguably) unnecessary.  One nice, and slightly subtle, reason to support this is that compile-time configs can take arbitrary code expressions as their default values rather than literals of the config's type.  For example, given:

```chapel
config const n = 100;
config const m = n / 2;
```

rather than just saying things like `-sm=25`, I can also say things like `-sm=n/4`.  This is described in the [users guide](https://chapel-lang.org/docs/users-guide/base/configs.html#compile-time-overrides-for-config-var-and-config-const).  In contrast, when setting config consts and vars at execution-time, only literal values of its type can be used (since we don't have an execution-time Chapel parser / interpreter).

The same is true of config params and types as well.  Essentially, think of the expressions that compile-time configs are set to as being text that is dropped into the Chapel program to initialize them.  (side note:  Unrelated to your issues, we recently tripped over a surprise in this behavior that we hadn't anticipated before... see issue #11783.

**Comment by bradcray** [comment]
I've got fixes for the first two issues in PR #11787 and #11789.  Both need review and testing, but were pretty straightforward.
